rafiki/db/database.py

```diff
@@ -285,54 +322,74 @@ def get_services(self, status=None):
     ####################################
 
     def create_model(self, user_id, name, task, model_file_bytes, 
-                    model_class, docker_image, dependencies):
+                    model_class, docker_image, dependencies, access_right):
         model = Model(
             user_id=user_id,
             name=name,
             task=task,
             model_file_bytes=model_file_bytes,
             model_class=model_class,
             docker_image=docker_image,
-            dependencies=dependencies
+            dependencies=dependencies,
+            access_right=access_right
         )
         self._session.add(model)
         return model
 
-    def get_models_of_task(self, task):
-        models = self._session.query(Model) \
-            .filter(Model.task == task).all()
+    def get_models_of_task(self, user_id, task):
 
+        public_models = self._session.query(Model) \
+            .filter(Model.task == task) \
+            .filter(Model.access_right == ModelAccessRight.PUBLIC) \
+            .all()
+
+        private_models = self._session.query(Model) \
+            .filter(Model.task == task) \
+            .filter(Model.access_right == ModelAccessRight.PRIVATE) \
+            .filter(Model.user_id == user_id) \
+            .all()
+
+        models = public_models + private_models
         return models
 
-    def get_model(self, id):
-        model = self._session.query(Model).get(id)
-        return model
+    def get_models(self, user_id):
+
+        public_models = self._session.query(Model) \
+            .filter(Model.access_right == ModelAccessRight.PUBLIC) \
+            .all()
+
+        private_models = self._session.query(Model) \
+            .filter(Model.access_right == ModelAccessRight.PRIVATE) \
+            .filter(Model.user_id == user_id) \
+            .all()
+
```

Comment:
Repeated logic - might want to add a private filter method `_filter_models(self, models, user_id)`

Reply:
Added a filter method